python/ray/_private/utils.py

```diff
@@ -350,16 +364,59 @@ def set_cuda_visible_devices(gpu_ids):
     Args:
         gpu_ids (List[str]): List of strings representing GPU IDs.
     """
-
     if os.environ.get(ray_constants.NOSET_CUDA_VISIBLE_DEVICES_ENV_VAR):
         return
+    set_visible_ids(gpu_ids, "CUDA_VISIBLE_DEVICES")
+
+
+def set_aws_neuron_core_visible_ids(core_ids):
+    """Set the NEURON_RT_VISIBLE_CORES environment variable.
+
+    Args:
+        core_ids (List[str]): List of strings representing core IDs.
+    """
+    if os.environ.get(ray_constants.NOSET_AWS_NEURON_VISIBLE_CORES_ENV_VAR):
+        return
+    set_visible_ids(core_ids, "NEURON_RT_VISIBLE_CORES")
 
-    global last_set_gpu_ids
-    if last_set_gpu_ids == gpu_ids:
+
+def set_visible_ids(visible_ids, env_var: str):
+    """Set the environment variable passed based on accelerator runtime.
+
+    Args:
+        visible_ids (List[str]): List of strings representing GPU IDs.
+        env_var: Environment variable to set based on GPU runtime.
+
+    """
+
+    global last_set_visible_ids
+    if last_set_visible_ids == visible_ids:
```

Comment:
will this collide GPU with NEURON?

Reply:
Good point, added more UT coverage and created separate global variables. From initialization perspective, we don't allow GPU and Neuron on one machine.